### PR TITLE
Task 5 Classification GFNet Pull Request - 47428364

### PR DESCRIPTION
**Pull request by s47428364**
Pull request to merge task 5 classification problem on the ADNI brain data using a GFNet.

Associated files are given are:
```
modules.py: Classes of the models for use in classification

dataset.py: Loads and processes the data

train.py: Trains a model on the dataset

predict.py: Predicts on the dataset from a trained mode

README.MD: Information about the task

readme_resources: Associated files for visualizing the README
```

More detail provide in the README

Caleb Clarke - 47428364

### CODE REVIEW

**Comment by wangzhaomxy** [comment]
**<This is an initial inspection, no action is required at this point.>**

**File Organizing:** Well-organized files.

**Problem Solving:**

- The algorithm can solve the problem but the model has been trained just 10 epochs, not reaching the ideal performance. 
- Accuracy in testing dataset: 0.70. 

**Model and functions:**

- Correctly uses PyTorch to construct the GFNet models and functions.
- With data augmentation.

**Code design:** Good.

**Code comment and docstring:**

- Good code comments 
- No function docstrings
- No head block nor code reference

**Difficulty:** Hard.

**Additional Comments:**

- Insufficient commits
- The model was trained just 10 epochs and saved one test acc result, so the acc plot in the report looks weird.
- The depiction of inputs is wrong, as the inputs should be images instead of a model. 
- No experiment analysis or conclusion is provided in the report.
- Vague user guide in ReadMe.

**Comment by aniketgupta17** [comment]
Observational Feedback

Pull Request:
Correctly created the Pull request from Topic Recognition Branch .
The pull request does not  include a **clear** description about the file structure .
No previous feedback incorporated . 

File Organizing: Well-organized files.

Commit Log:
Commit messages could be more progressive for the Recognition Problem solved using 4 files . 
Commits are not  regularly made.

Documentation:
Incomplete README.

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Adequate design and implementation. </td><td>-1</td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>No Header blocks. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>OK solution to problem. Code needs fixing for training.</td><td>-3</td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting present. </td><td></td></tr>
	<tr><td></td><td>No Data leakage found. Uses Data Aug</td><td></td></tr>
	<tr><td></td><td>Difficulty : Hard.  GFNet (Hard Difficulty)</td><td></td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Some/Adequate Meaningful commit messages. </td><td>-1</td></tr>
	<tr><td></td><td>Some/Adequate Progressive commits. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Acceptable. Incomplete.</td><td>-3</td></tr>
	<tr><td></td><td>Model/technical explanation :Acceptable. </td><td>-1</td></tr>
	<tr><td></td><td>Description and Comments :Good. </td><td></td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF Checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). </td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is good. </td><td></td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-11</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by shakes76** [comment]
Training issues found. Does not solve problem, inconsistencies. -2, error in code -1, results can't be obtained properly -2
